File: STRING/PYTHON/naive.py
# Naive String Matching Algorithm - A brute force Approach

def naiveStringMatchingAlgo(text,pattern):
    result = []
    n = len(text)
    m = len(pattern)

    for i in range(n - m + 1):
        j = 0
        while j < m and pattern[j] == text[i+j]:
            j += 1

        if j == m:
            result.append(i)

    return result if result else -1

text = 'abcabaabcabaa'
pattern = 'abaa'
print(naiveStringMatchingAlgo(text,pattern))

File: STRING MATCHING/JAVA/naive.java
// Naive String Matching Algorithm - A brute force Approach
import java.util.*;

public class naive{

    public static List<Integer> NaiveStringAlgo(String text, String pattern) {

        List<Integer> result = new ArrayList<>();
        int n = text.length();  // textLength 
        int m = pattern.length(); // patternlength

        for (int i = 0; i <= n -m; i++) {
            int j = 0;
            while(j < m && pattern.charAt(j) == text.charAt(i+j)){
                j += 1;
            }

            if (j == m) {
                result.add(i); 
            }
        }
    return result;
    }

    public static void main(String[] args) {
        String text = "THIS IS A TEST TEXT FOR YOU PROGRAM TO TEST";
        String pattern = "TEST";

        List<Integer> indices = NaiveStringAlgo(text, pattern);

        for (int index : indices) {
            System.out.println("Pattern found at index " + index);
        }
    }
}